model/client4.go

```diff
@@ -4267,94 +4248,94 @@ func fileToMultipart(data []byte, filename string) ([]byte, *multipart.Writer, e
 
 // UploadSamlIdpCertificate will upload an IDP certificate for SAML and set the config to use it.
 // The filename parameter is deprecated and ignored: the server will pick a hard-coded filename when writing to disk.
-func (c *Client4) UploadSamlIdpCertificate(data []byte, filename string) (bool, *Response) {
+func (c *Client4) UploadSamlIdpCertificate(data []byte, filename string) (bool, *Response, error) {
 	body, writer, err := fileToMultipart(data, filename)
 	if err != nil {
-		return false, &Response{Error: NewAppError("UploadSamlIdpCertificate", "model.client.upload_saml_cert.app_error", nil, err.Error(), http.StatusBadRequest)}
+		return false, nil, NewAppError("UploadSamlIdpCertificate", "model.client.upload_saml_cert.app_error", nil, err.Error(), http.StatusBadRequest)
 	}
 
-	_, resp := c.DoUploadFile(c.GetSamlRoute()+"/certificate/idp", body, writer.FormDataContentType())
-	return resp.Error == nil, resp
+	_, resp, err := c.DoUploadFile(c.GetSamlRoute()+"/certificate/idp", body, writer.FormDataContentType())
+	return true, resp, err //TODO
```

Comment:
Unhandled TODO? We need to check for `err != nil` here to determine the bool return value. As an improvement, I'd suggest to remove the bool param altogether, since it is derivable from the error variable which we are already returning.

Reply:
Good suggestion. Added that in 41afad1.